### PR TITLE
Refactor the project layout

### PR DESCRIPTION
**Pull request by huachaohuang**
This is a part of #59 . Other parts will be added as v0.2 proceed.

### CODE REVIEW

File: src/engula-kernel/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[package]
+name = "engula-kernel"
+version = "0.1.0"
+edition = "2021"
```
Comment (tisonkun) [inline src/engula-kernel/Cargo.toml:4]: add a brief description?

File: src/engula-kernel/manifest/Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 [package]
-name = "engula-platform-storage"
+name = "manifest"
 version = "0.1.0"
 edition = "2021"
+publish = false
```
Comment (tisonkun) [inline src/engula-kernel/manifest/Cargo.toml:5]: add a brief description?

File: src/engula-platform/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[package]
+name = "engula-platform"
+version = "0.1.0"
+edition = "2021"
```
Comment (tisonkun) [inline src/engula-platform/Cargo.toml:4]: add a brief description?

File: src/engula-platform/storage/Cargo.toml [hunk 4]
```diff
@@ -1,7 +1,8 @@
 [package]
-name = "engula-kernel-manifest"
+name = "storage"
 version = "0.1.0"
 edition = "2021"
+publish = false
```
Comment (tisonkun) [inline src/engula-platform/storage/Cargo.toml:5]: add a brief description?

File: src/engula/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "engula"
+version = "0.1.0"
+edition = "2021"
```
Comment (tisonkun) [inline src/engula/Cargo.toml:4]: add a brief description?

File: src/engula/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (tisonkun) [inline src/engula/Cargo.toml:6]:
```suggestion
```

remove redundant template context

**Review by tisonkun — COMMENTED** [review]
Nice work. Comments inline.

**Comment by huachaohuang** [comment]
@tisonkun How about we decide those fields in Cargo.toml when we settle things down and before publishing? #61

**Review by tisonkun — APPROVED** [review]
@huachaohuang make senses. You can add a todo item in the roadmap which we will check before release.

**Pull request merged**